### PR TITLE
Feat/6 prototype for integration of orbital analysis

### LINKED ISSUE #6: Prototype for integration of orbital analysis
## Checklist before submitting a feature request

- [X] I have reviewed [CONTRIBUTING.md](https://github.com/opencaesar/oml-vision/blob/master/CONTRIBUTING.md) document
- [X] I have done a quick search on the issue tracker to check if the feature request is not redundant

## Is your feature request related to a problem? If yes, describe the problem

It's nice to demonstrate the integration of mathematical analysis capabilities with the OML descriptive model.  

The statement of work or the business need for this ticket is to create a Quarto notebook which executes Python and R simulation code on the OML descriptive model.  

## Related Tasks

- [ ] Saving the OML descriptions after the simulations are run will be handled in #11 
- [ ] #13
- [ ] #14

## Describe the desired feature

[Clearly and concisely describe the functionality that you would like to see]

## Additional context

[Any additional context]

### PR DESCRIPTION
**Pull request by UTNAK**
## Checklist before submitting a merge request

- [x] I have reviewed [CONTRIBUTING.md](https://github.com/opencaesar/oml-vision/blob/master/CONTRIBUTING.md) document
- [x] All commits have been signed-off for the Developer Certificate of Origin. See [here](https://github.com/opencaesar/oml-vision/blob/master/CONTRIBUTING.md#developer-certificate-of-origin).
- [x] I have added my contribution to [CHANGELOG.md](https://github.com/opencaesar/oml-vision/blob/master/CHANGELOG.md)
- [x] I have documented my code in the [OML Vision Docs](http://www.opencaesar.io/oml-vision-docs/)

## Description of contribution

Using the quarto notebook as an analysis engine, this PR demonstrate the integration of mathematical analysis using Python and R with OML Descriptive models.

Close #6

### Prepare Environment

Before you run simulations on your local computer, you need to setup simulation environments.

- [ ] Install and Anaconda Python
- [ ] Install R and Quarto [instruction is here](https://quarto.org/docs/get-started/)
- [ ] Setup VSCode with quarto [instruction is here](https://quarto.org/docs/get-started/hello/vscode.html)

### Testing

In VSCODE, 
- [ ] Clone codes from github.
- [ ] Run all Cells in SetUpEnvironment.qmd.
- [ ] Run Build and Load tasks of gradle using oml-vision.
- [ ] Run all Cells in analysisShell.qmd. (* you need to select python interpreter as py393)
- [ ] See the results, calculation of R and Python are correct. The expected outputs are shown in [here](https://github.com/UTNAK/kepler16b-using-imce-vocabulary/pull/12/files#diff-64c87f50ab1368586bfa462083ad394dcbafa5e1cea4876d1caefc222f8b8b1b)

### CODE REVIEW

File: src/analysis/analysisShell.qmd [hunk 1]
```diff
@@ -0,0 +1,733 @@
...
+```
+
+```{r}
+
+df_table <- df_parameters %>% 
+  mutate(id=1) %>%
+  pivot_longer(!id, names_to = "parameter", values_to = "value") %>%
+  select(-id)
+
+datatable(df_table, options = list(autoWidth = TRUE, pageLength = -1))
```
Comment (UTNAK) [inline src/analysis/analysisShell.qmd:306]: check the parameters are queried and stored in this dataframe.

File: src/analysis/analysisShell.qmd [hunk 2]
```diff
@@ -0,0 +1,810 @@
...
+          c2_instancename=df_config$c1_instancename[1],
+          c2_id=df_config$c1_id[1],
+          c2_name=df_config$c1_name[1],
+          c2_mass=df_config$c1_mass[1],
+          c1_type=df_config$c1_type[1],
+          c2_label=NA,
+          c1_label=df_config$c1_label[1],  
+          .before = 1)
+
+plotCollapsibleTreeFromDataframe(df_vis, palette="BluYl",
```
Comment (UTNAK) [inline src/analysis/analysisShell.qmd:351]: Plot result is expected like this

image

File: src/analysis/analysisShell.qmd [hunk 3]
```diff
@@ -0,0 +1,810 @@
...
+from astropy import units as u
+
+analysisRocket = analysisOrbit.hohmanTransfer(orbit_init=orbit_init, orbit_target=orbit_target)
+# analysisRocket = analysisOrbit.hohmanTransfer(orbit_init=r.initOrbit, orbit_target=r.targetOrbit, m_init=r.m_dry, I_sp=r.I_sp)
+# analysisRocket = analysisOrbit.hohmanTransfer(orbit_init=400, orbit_target=35786, m_init=5000, I_sp=300)
+
+total_delta_v = analysisRocket.calculate_delta_v()
+
+
+print(f"Total delta-v: {total_delta_v}")
```
Comment (UTNAK) [inline src/analysis/analysisShell.qmd:512]: Expected results are

`>>>  print(f"Total delta-v: {total_delta_v}")
Total delta-v: 3853.959363102248 m / s
>>>  print(f"Required fuel mass: {analysisRocket.m_fuel}\n final: {analysisRocket.m_final}\n init: {analysisRocket.m_init}")
Required fuel mass: 0.0
 final: 0.0
 init: 1000.0`

**Comment by pogi7** [comment]
@UTNAK I'm going to add these instructions into the dev container so recreating this environment is more consistent for developers and users.

Anaconda is found here:
https://github.com/devcontainers/images/tree/main/src/anaconda

R and Quarto that works with VSCode found here:
https://jamesgoldie.dev/writing/dev-containers-in-r/

> Prepare Environment
> 
> Before you run simulations on your local computer, you need to setup simulation environments.
> 
>  Install and Anaconda Python
>  Install R and Quarto [instruction is here](https://quarto.org/docs/get-started/)
>  Setup VSCode with quarto [instruction is here](https://quarto.org/docs/get-started/hello/vscode.html)

File: src/analysis/analysisShell.qmd (hunk 1 above)
Comment (pogi7) [reply src/analysis/analysisShell.qmd:306]: @UTNAK I see the same thing in my codespace

image

File: src/analysis/analysisShell.qmd (hunk 2 above)
Comment (pogi7) [reply src/analysis/analysisShell.qmd:351]: @UTNAK I see the same thing in my codespace

image

File: src/analysis/analysisShell.qmd [hunk 4]
```diff
@@ -0,0 +1,810 @@
...
+
+df_table$componenttype <- factor(df_table$componenttype)
+
+datatable(df_table, options = list(autoWidth = FALSE, pageLength = -1), filter = list(
+  position = 'top', clear = FALSE
+))
+
+
+```
+
```
[the remark this answers is not in the document]
Comment (pogi7) [reply src/analysis/analysisShell.qmd:467]: @UTNAK I see the same thing on my codespace

image

**Comment by UTNAK** [comment]
> @UTNAK Python section of codespace is not working.
> 
> Getting an `OSError` after running Quarto notebook to calculate delta v section


This is due to the load of 

from poliastro.twobody import Orbit

in src/analysis/analysisOrbit.py

I found that the this code works in python at terminal.
In quarto notebook, this doesn't work

File: src/analysis/analysisShell.qmd (hunk 3 above)
Comment (pogi7) [reply src/analysis/analysisShell.qmd:508]: @UTNAK I see the same thing
>>>  print(f"Total delta-v: {total_delta_v}")
Total delta-v: 3853.959363102248 m / s

File: src/analysis/analysisShell.qmd [hunk 5]
```diff
@@ -0,0 +1,810 @@
...
+df_table <- df_parameters %>% 
+  mutate(id=1) %>%
+  pivot_longer(!id, names_to = "parameter", values_to = "value") %>%
+  select(-id)
+
+datatable(df_table, options = list(autoWidth = TRUE, pageLength = -1))
+
+```
+
```
[the remark this answers is not in the document]
Comment (pogi7) [reply src/analysis/analysisShell.qmd:548]: @UTNAK I see the same thing
image

**Review by pogi7 — APPROVED** [review]
@UTNAK looks good to me!

**Pull request merged**